### DISCUSSION TITLE
Roadmap 2023-2025

### DISCUSSION THREAD

**Discussion by LucyKershaw** [OSIPI/DCE-DSC-MRI_CodeCollection#118] [General]
Hello all,

We were having a chat about the main pieces of work that we've proposed for the next 2 years.  Here are the two proposals:
1) Automation of code submission and testing incorporating the ideas from Yu Sun.  The outcome will be to reduce the amount of manual tasks currently needed for the testing of new submissions.  We could also allow code submission in other (open) languages.
2) User-friendly pipeline. The outcome will be an interactive notebook that shows the results of model fitting for our existing DRO data (with ground truth) depending on the selection of different AIFs for the most basic version.  If there's time we can add different models, different bolus arrival time estimates, and potentially allow users to add their own data and compare between processing pipelines.  The final goal would be for the user to be able to download a chosen pipeline that contains all the code required for analysis, except the image processing input/output methods.

Based on our meeting last week, here's the proposed roadmap:
Feb 2023: 
- Start automation of code submission and testing process
- Start user-friendly pipeline (notebook)

ISMRM 2023 June 2023: 
- Complete wrapper functions for code testing
- Finalise design of pipeline (software, scope, plans for extensions)

ISMRM 2024 abstract deadline Nov 2023: 
- Submit abstract about automated code testing 
- Submit abstract with basic pipeline notebook

ISMRM 2024 May 2024:
- Finish code testing and submission automation
- Finish pipeline notebook with some extras if possible

ISMRM 2025 Submit manuscript
- One manuscript on either of the main strands above - TBC

Time is tight, so if you have any comments, please let us know by the 14th Nov 

Lucy and Sirisha

**Comment by mjt320** [comment]
Apologies, only just seen this. I think my notifications are getting lost.
It sounds good to me.
Two thoughts:
1. do we mean actual automation or are we talking about streamlining the process?
2. should we also aim to expand the code content of the repository - there are some gaps and we're light on DSC content

**Reply by stadmill** [reply]
Thanks Michael. We did mean actual automation of the code submission and testing. In my imagination, the way it would work is that when new code comes in, the process is triggered and all the steps through to publishing the plots on the website are automatically completed, without the need for manually editing the tests or json files etc. I think for something completely new, like DSC code, we will still need to manually write test cases, but anything after should be automatic. Having said that, we don't really have a plan on how we will achieve this. I think the definition of the scope will be Milestone 1 (Feb 2023, as per the roadmap).  

We do need to expand the DSC content, but I don't think it is a specific aim.

**Comment by plaresmedima** [comment]
Hi,

I don't really have any comments other than that this sounds like a good
plan! Addresses the priority of consolidating rather than expanding, and
should be doable in the time frame. I suppose you will remain open to new
submissions though? This does not have to be in the timeline but if new
submissions come will you still integrate them?

Prof. Steven Sourbron, PhD
Chair in Medical Imaging Physics
University of Sheffield, UK
*https
<https://www.sheffield.ac.uk/medicine/people/iicd/steven-sourbron>://www.sheffield.ac.uk/medicine/people/iicd/steven-sourbron
<https://www.sheffield.ac.uk/medicine/people/iicd/steven-sourbron>*
*Sheffield abdominal imaging research page*
<https://www.sheffield.ac.uk/medicine/research/research-themes/medical-imaging/medical-imaging-research/abdominal-imaging>
*Open source initiative in perfusion imaging* <http://www.osipi.org>
*Magnetic resonance imaging biomarkers for chronic kidney disease
<https://renalmri.org/>*

**Reply by stadmill** [reply]
Yes new submissions will still be integrated. The idea is to keep all things going as they are while improving/automating different aspects of the submission and testing process.

**Comment by petravanhoudt** [comment]
Same apologies here, my notifications were also not set properly apparently! Maybe it is good to make people in the TF aware of this?
BTW, I saw that there is also a TF2.3 team created on github: https://github.com/orgs/OSIPI/teams/task-force-2-3. Should we remove this to avoid confusion?

wrt the roadmap, I like you suggestions! I like it that you want to focus on both streamlining the process (reduce the amount of copy-paste :)) as well as create a user-friendly example of a pipeline. This will show how people can use the code repository and the future possibilities of what you are trying to set up.

**Reply by stadmill** [reply]
I'll email and let people know about the notifications. 

Did you get rid of the other team page? I get a 404 error now.

**Reply by petravanhoudt** [reply]
It was still there.  I have just deleted it. There were no messages there, so nothing got lost.

**Comment by LucyKershaw** [comment]
Hello all,

You should (shortly) be getting a link to our section of the roadmap so you can make comments.  Anyone who doesn't get this, please email me directly.

Lucy

**Reply by LucyKershaw** [reply]
Thanks to everyone who made comments, we've copied a clean version into the roadmap document but also kept the version that's just our for reference.  Now we just have to follow it....

**Comment by JonathanArvidsson** [comment]
Hi guys,

I just wanted to chip in with a comment in regard to the pipeline.

Make sure to checkout [NiPype](https://nipype.readthedocs.io/en/latest/). 

Once wrappers to functions are made, building and editing pipelines is really straight forward. This also enables batch processing out of the box as well as automated illustrations of the pipeline.

since wrappers for the regular imaging toolboxes are already implemented, adding i/o steps and spatiotemporal registrations is basically copy/paste.

Jonathan
